qiskit/quantum_info/synthesis/two_qubit_decompose.py

```diff
@@ -1403,6 +1403,98 @@ def num_basis_gates(self, unitary):
         return np.argmax([trace_to_fid(traces[i]) * self.basis_fidelity**i for i in range(4)])
 
 
+class TwoQubitDecomposeUpToDiagonal:
+    """
+    Class to decompose two qubit unitaries into the product of a diagonal gate
+    and another unitary gate which can be represented by two CX gates instead of the
+    usual three. This can be used when neighboring gates commute with the diagonal to
+    potentially reduce overall CX count.
+    """
+
+    def __init__(self):
+        self.sy = np.array([[0, -1j], [1j, 0]])
+        self.sz = np.array([[1, 0], [0, -1]])
+        self.sysy = np.kron(self.sy, self.sy)
+
+    def _u4_to_su4(self, u4):
+        from scipy import linalg as la
```

Comment:
I did  this as it was also done in the `TwoQubitWeylDecomposition.__new__()`. Perhaps it reduces load time in some cases.

Reply:
Yeah, for a bunch of the scipy modules like this runtime imports are used to avoid the import cost on the root `import qiskit` call. Scipy imports are relatively heavy compared to the import time of the rest of qiskit.  Unless this particular method is particularly heavily used and quick so that adding a dict lookup on each call is too expensive, which I don't think it is in this case.